packages/account/src/providers/utils/handle-gql-error-message.ts

```diff
@@ -0,0 +1,20 @@
+import { ErrorCode, FuelError } from '@fuel-ts/errors';
+import type { GraphQLError } from 'graphql';
+
+export enum GqlErrorMessage {
+  NOT_ENOUGH_COINS = 'not enough coins to fit the target',
+}
+
+export const handleGqlErrorMessage = (errorMessage: string, rawError?: GraphQLError) => {
```

Comment:
Should the `rawError` be mandatory instead of optional? Every gql error must have an accompanying `GraphQLError` error.

Reply:
I kept it optional because I'm only getting the error _message_ from the GraphQL subscription implementation that we have :/

Reply:
This error will never happen in `FuelGraphqlSubscriber` because we're not using the `submitAndAwait` endpoint anymore; we only use `statusChange`, which can't error with `NOT_ENOUGH_FUNDS` because it works with already-submitted transactions. I'm for completely removing the work done in `FuelGraphqlSubscriber` because of that and thinking about the wider problem in #2467, given that the scope of this PR/issue is only related to `NOT_ENOUGH_FUNDS`.

Reply:
If you still want to keep it in `FuelGraphqlSubscriber` - which I'm also fine with - you can change the `rawError` parameter's type to be `FuelError['rawError']` which is `unknown` and then you can pass in the whole error object in `FuelGraphqlSubscriber` as well:
```ts
for (const error of errors) {
  handleGqlErrorMessage(error.message, error);
}
```

Reply:
> I'm for completely removing the work done in FuelGraphqlSubscriber because of that and thinking about the wider problem in https://github.com/FuelLabs/fuels-ts/issues/2467

@nedsalk could you just clarify what you mean by this? Maybe on the #2467 would be helpful :)

Reply:
@nedsalk Thanks for the insight. Surely having the logic contained within `handleGqlErrorMessage` is keeping it centralized?

Reply:
Indeed it is, but both subscriptions and non-subscriptions are going through the same fetch function so there _might_ be room for restructuring the code and using `handleGqlErrorMessage` only in that fetch function as opposed to the current usage within both `FuelGraphqlSubscriber` and `GraphqlRequest`.